Can't build astro if project path contains spaces on MacOS

### Astro Info

```block
> astro info

Astro                    v5.0.0-beta.12
Node                     v18.20.4
System                   macOS (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/mdx
                         @astrojs/sitemap
```


### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

This might be an NPM or a Vite problem but I just started using Astro and it refuses to build if the project path contains a space on MacOS: for example, ".../new website" does not build, while ".../website" does.

The error is as follows:

astro vscode [ERROR] [vite] cannot find entry point module '.../Documents/new%20website/Site/node_modules/@astrojs/mdx/dist/server.js'.

It goes away if I change the path to the project to this: ".../Documents/Website/Site/node_modules/@astrojs/mdx/dist/server.js"

It was easy to fix for me, but I imagine if for example someone's username contains a space then they are in deep trouble.

### What's the expected result?

Astro building my site.

### Link to Minimal Reproducible Example

should be trivial to reproduce - in my case I just used the blog template and created a scratch project under a path with a space in the pathname.

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/packages/vite/src/module-runner/utils.ts
+++ b/packages/vite/src/module-runner/utils.ts
@@ -7,7 +7,7 @@ export function normalizeAbsoluteUrl(url: string, root: string): string {
   // file:///C:/root/id.js -> C:/root/id.js
   if (url.startsWith('file://')) {
     // 8 is the length of "file:///"
-    url = url.slice(isWindows ? 8 : 7)
+    url = decodeURI(url.slice(isWindows ? 8 : 7))
   }
 
   // strip root from the URL because fetchModule prefers a public served url path

--- a/packages/vite/src/node/ssr/__tests__/fixtures/file-url/test space.js
+++ b/packages/vite/src/node/ssr/__tests__/fixtures/file-url/test space.js
@@ -1,0 +1,1 @@
+export const msg = 'works'

--- a/packages/vite/src/node/ssr/__tests__/fixtures/file-url/test.js
+++ b/packages/vite/src/node/ssr/__tests__/fixtures/file-url/test.js
@@ -1,0 +1,1 @@
+export const msg = 'works'

--- a/packages/vite/src/node/ssr/__tests__/ssrLoadModule.spec.ts
+++ b/packages/vite/src/node/ssr/__tests__/ssrLoadModule.spec.ts
@@ -237,3 +237,17 @@ test('json', async () => {
   )
   expect(json?.code.length).toMatchInlineSnapshot(`61`)
 })
+
+test('file url', async () => {
+  const server = await createDevServer()
+
+  const mod = await server.ssrLoadModule(
+    new URL('./fixtures/file-url/test.js', import.meta.url).href,
+  )
+  expect(mod.msg).toBe('works')
+
+  const modWithSpace = await server.ssrLoadModule(
+    new URL('./fixtures/file-url/test space.js', import.meta.url).href,
+  )
+  expect(modWithSpace.msg).toBe('works')
+})

